rllib/agents/trainer.py

```diff
@@ -1823,10 +1989,12 @@ def _try_recover(self):
             raise RuntimeError(
                 "Not enough healthy workers remain to continue.")
 
-        logger.warning("Recreating execution plan after failure")
+        logger.warning("Recreating execution plan after failure.")
         workers.reset(healthy_workers)
-        self.train_exec_impl = self.execution_plan(
-            workers, self.config, **self._kwargs_for_execution_plan())
+        if self.train_exec_impl is not None:
```

Comment:
hmm I wonder why we need to check this here ...?

Reply:
We do. What if the user does not provide an `execution_plan` but implements `step()` herself?

Reply:
I see. so in this case user does her own init() and step(), and we can't really recover anything for her here. that is quite interesting.
can you please help comment this.
thanks.